plugins/org.yakindu.sct.doc.user/src/user-guide/embedded_systems_integration_guide.textile

```diff
@@ -0,0 +1,281 @@
+	statemachine_enter();
+
+	// While loop
+	while(!statemachine_isFinal()) {
+		if(hasInEvents) {
+			handleInEvents();
+		}
+		
+		if(hasTimeEvents) {
+			handleTimer();
+		}
+		
+		if(isCycleBased && elapsedTime >= cyclePeriod) {
+			elapsedTime = 0;
+			statemachine_runCycle();
+		}
+		
+		if(hasOutEvents) {
+			handleOutEvents();
+		}
+		
+		if(usingSleepMode) {
+			goToSleep()
+			// wait for Interrupt
+		}
+	}
+}
+p. 
+
+h3(#handle-in-events). Handle In Events 
+
+For now on, there are two different ways of how the events should be interact on the embedded system. One design pattern is polling. The other one is the usage of interrupts.
+
+A design using polling cyclic updates the status of the inputs and raise the according event.
+
+bc.. 
+void handleInEvents() {
+	if(readGPIO(1)) {
+		statemachine_raise_inEvent1();
+	}
+	if(readGPIO(2)) {
+		statemachine_raise_inEvent2();
+	}
+}
+p. 
+
+As already described in the Interrupt section, a design using interrupts recommends bool flags as storage of the event.
+
+bc.. 
+void handleInEvents() {
+	if(inEvent1) {
+		statemachine_raise_inEvent1();
+		inEvent1 = false;
+	}
+	if(inEvent2) {
+		statemachine_raise_inEvent2(inEvent2Value);
+		inEvent2 = false;
+	}
+}
+
+Sensor1_ISR{
+	inEvent1 = true;
+}
+
+Sensor2_ISR{
+	inEvent2 = true;
+	inEvent2Value = readSensor2();
+}
+p. 
+
+h3(#handle-timer). Handle Timer
+
+Handling the timer is more comprehensive. Each time event of the state machine must be updated and handled, because in detail there are nothing else than a in event. This means that for every timed trigger there must be one in event. Therefore, a timer service is provided in the examples. This timer service must be updated with the elapsed time since the its last call, which leads to the next implementation point: Determining the time.
+On a pure bare-metal implementation there is no such thing as elapsed time. Everything depends on cycles. 
+
+Imagine a microcontroller running with a clock of 16MHz. Running one instruction needs 1/16MHz = 625ns. A delay() function 1000000 would need 6,25 ms.A first approach updating the timer could be:
+
+bc.. 
+void main() {
+	while(true) {
+		delay(1000000);
+		updateTimer(6,25);
+		runCycle();
+	}
+}
+p. 
+
+But using this implementations got two big drawbacks:
+* the delay function is blocking
+* the execution time of the updateTimer and runCycle are not taking into account
+
+A much more precise design is using timer interrupts, which can generate a periodic interrupt ever x cycles:
+
+bc.. 
+void handleTimer() {
+	if(updateTimerFlag) {
+		updateTimer(TIMER_TICK_MS);
+		updateTimerFlag = false;
+	}
+}
+
+#define TIMER_TICK_MS 32 // 32 ms for example
+Timer_ISR{ 
+	updateTimerFlag = true;
+}
+p. 
+
+Some microcontrollers, like the Arduino, support functions like millis(), which are in fact using the a similar method, but storing the elapsed time since starting the device. This time can be used to determine the elapsed time between two cycles in a loop:
```

Comment:
```suggestion
Some microcontrollers, like the Arduino, support functions like millis(), which are in fact using a similar method, but storing the elapsed time since starting the device. This time can be used to determine the elapsed time between two cycles in a loop:
```